src/lib/core/TLVBackingStore.h

```diff
@@ -155,6 +155,8 @@ class DLL_EXPORT TLVBackingStore
      *
      */
     virtual CHIP_ERROR FinalizeBuffer(TLVWriter & writer, uint8_t * bufStart, uint32_t bufLen) = 0;
+
+    virtual bool IsSafeToReserve() { return false; }
```

Comment:
Document what exactly this guarantees?

Reply:
After reading through all the comments, and thinking about this some more I think a better name is GetNewBufferWillAlwaysFail(), or `UnderlyingBufferCanBeExtended()`.

If anyone has a more succinct  recommendation I would gladly hear it